tornado/ioloop.py

```diff
@@ -1123,17 +1160,20 @@ def time(self):
     def call_at(self, deadline, callback, *args, **kwargs):
         timeout = _Timeout(
             deadline,
-            functools.partial(stack_context.wrap(callback), *args, **kwargs),
+            partial(wrap(callback), *args, **kwargs),
             self)
         heapq.heappush(self._timeouts, timeout)
         return timeout
 
+    @cython.locals(timeout = _Timeout)
     def remove_timeout(self, timeout):
         # Removing from a heap is complicated, so just leave the defunct
         # timeout object in the queue (see discussion in
         # http://docs.python.org/library/heapq.html).
         # If this turns out to be a problem, we could add a garbage
         # collection pass whenever there are too many dead timeouts.
+        if timeout is None:
```

Comment:
If timeout is None, this will already raise an exception. Why did you add the additional check?

Reply:
`timeout` is now declared as an extension type, which means dereferencing None -> segfault. We thus need explicit checks.